src/cairo.jl

```diff
@@ -80,12 +82,21 @@ function render(c::CairoContext,g::GraphViz.Graph; context = default_context[],
 end
 
 function cairo_render(g::GraphViz.Graph; context = default_context[], format="julia:cairo")
-    global last_surface
+    global last_surface, format_placeholder
     GraphViz.add_julia_cairo!(context)
     if !g.didlayout
         error("Must call layout before calling render!")
     end
-    ccall((:gvRenderContext,libgvc),Cint,(Ptr{Cvoid},Ptr{Cvoid},Ptr{UInt8},Ptr{Cvoid}),context.handle,g.handle,format,C_NULL)
+
+    empty!(format_placeholder)
+    append!(format_placeholder, format)
+    push!(format_placeholder, UInt8(0))
+    format_cstring = pointer(format_placeholder)
+
+    ret = ccall((:gvRenderContext,libgvc),Cint,(Ptr{Cvoid},Ptr{Cvoid},Ptr{UInt8},Ptr{Cvoid}),context.handle,g.handle,format_cstring,C_NULL)
```

Comment:
This is all not gc-safe. Just declare this as `Cstring` rather than `Ptr{UInt8}` and let julia worry about the string conversion.